lntest/itest/lnd_mpp_test.go

```diff
@@ -0,0 +1,353 @@
+// payment using multiple shards across different paths, by using SendToRoute.
+func testSendToRouteMultiPath(net *lntest.NetworkHarness, t *harnessTest) {
+	ctxb := context.Background()
+
+	// To ensure the payment goes through seperate paths, we'll set a
+	// channel size that can only carry one shard at a time. We'll divide
+	// the payment into 3 shards.
+	const (
+		paymentAmt = btcutil.Amount(300000)
+		shardAmt   = paymentAmt / 3
+		chanAmt    = shardAmt * 3 / 2
+	)
+
+	// Set up a network with three different paths Alice <-> Bob.
+	//              _ Eve _
+	//             /       \
+	// Alice -- Carol ---- Bob
+	//      \              /
+	//       \__ Dave ____/
+	//
+	// We'll send shards along three routes from Alice.
+	sendRoutes := [][]string{
+		{"carol", "bob"},
+		{"dave", "bob"},
+		{"carol", "eve", "bob"},
+	}
+
+	nodes := make(map[string]*lntest.HarnessNode)
+	nodes["alice"] = net.Alice
+	nodes["bob"] = net.Bob
+
+	// Keep a list of all our active channels.
+	var networkChans []*lnrpc.ChannelPoint
+
+	// Open channels to/from the three new nodes.
+	nodeNames := []string{"carol", "dave", "eve"}
+	for _, name := range nodeNames {
+		node, err := net.NewNode(name, nil)
+		if err != nil {
+			t.Fatalf("unable to create new nodes: %v", err)
+		}
+		defer shutdownAndAssert(net, t, node)
+
+		nodes[name] = node
+
+		// All nodes will have a channel to Bob.
+		ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+		if err := net.ConnectNodes(ctxt, node, net.Bob); err != nil {
+			t.Fatalf("unable to connect %v to alice: %v", name, err)
+		}
+		ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+		err = net.SendCoins(ctxt, btcutil.SatoshiPerBitcoin, node)
+		if err != nil {
+			t.Fatalf("unable to send coins to %v: %v", name, err)
+		}
+		ctxt, _ = context.WithTimeout(ctxb, channelOpenTimeout)
+		chanPointBob := openChannelAndAssert(
+			ctxt, t, net, node, net.Bob,
+			lntest.OpenChannelParams{
+				Amt: chanAmt,
+			},
+		)
+
+		defer func() {
+			ctxt, _ := context.WithTimeout(ctxb, channelCloseTimeout)
+			closeChannelAndAssert(
+				ctxt, t, net, node, chanPointBob, false,
+			)
+		}()
+
+		networkChans = append(networkChans, chanPointBob)
+
+		// Set up channels coming into the node.
+		var from *lntest.HarnessNode
+		switch name {
+
+		// Eve will have a channel from Carol.
+		case "eve":
+			from = nodes["carol"]
+
+		// The other nodes will have a channel from Alice.
+		default:
+			from = net.Alice
+		}
+
+		ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+		if err := net.ConnectNodes(ctxt, from, node); err != nil {
+			t.Fatalf("unable to connect %v to %v: %v", from.Name(),
+				name, err)
+		}
+
+		// Since the channel Alice-> Carol will have to carry two
+		// shards, we make it larger.
+		amt := chanAmt
+		if name == "carol" {
+			amt += shardAmt
+		}
+
+		ctxt, _ = context.WithTimeout(ctxb, channelOpenTimeout)
+		chanPointAlice := openChannelAndAssert(
```

Comment:
When continuing work/debugging on the mpp payment itest, I found it difficult to figure out what the exact channel parameters are. In https://github.com/lightningnetwork/lnd/pull/3967, I modified the itest commit to open a set of channels with explicit parameters. I think that would also be beneficial for this test.

Reply:
Cool, done: b74089ca9